genetic_algorithm/one_max.py

```diff
@@ -0,0 +1,233 @@
+from collections.abc import Callable
+
+from numpy.random import rand, randint
+
+"""
+This program performs genetic evolution on a list of bits.
+Its goal is to transform every bit to a 1, using mutations and crossovers.
+https://en.wikipedia.org/wiki/Genetic_algorithm
+"""
+
+
+def goal(individual: list[int]) -> int:
+    """
+    Calculate the fitness score for an individual in the genetic evolution algorithm.
+
+    The goal of the algorithm is to transform every bit to a 1, and the fitness
+    score is the negation of the sum of the bits.
+
+    Parameters:
+    - individual (list[int]): A list representing an individual in the population.
+
+    Returns:
+    - int: The fitness score for the individual.
+
+    Examples:
+    >>> goal([1, 1, 1, 1, 1, 1, 1, 1])
+    -8
+    >>> goal([0, 0, 0, 0, 0, 0, 0, 0])
+    0
+    >>> goal([1, 0, 1, 0, 1, 0, 1, 0])
+    -4
+    """
+    return -sum(individual)
+
+
+def selection(population: list, scores: list[int], k: int) -> list:
+    """
+    Select individuals from the population based on tournament selection.
+
+    Tournament selection involves randomly selecting k individuals from the
+    population and choosing the one with the highest fitness score.
+
+    Parameters:
+    - population (list): The list of individuals in the population.
+    - scores (list[int]): The corresponding fitness scores for each individual.
+    - k (int): The number of individuals to select in each tournament.
+
+    Returns:
+    - list: The selected individual from the tournament.
+
+    Examples:
+    >>> population = [[1, 0, 1, 1, 0], [0, 1, 0, 0, 1], [1, 1, 1, 0, 1]]
+    >>> scores = [-2, -3, -1]
+    >>> selection(population, scores, 2) in population
+    True
```

Comment:
The function signature says we are going to return a `list` but we return a `bool`.